Mozilla All Hands 2017

arp95 opened the pull request:
fixes #149

Sample app: https://arp95.github.io/MozillaAllHands_17/

started working with the mozilla event, which the schedule published here: https://sanfranciscoallhandsjune2017.sched.com/

arp95 commented:
@iamareebjamal @mahikaw @Shailesh351 @shreyasnbhat
please review now.

arp95 commented:
Guys please review. It needs to be merged fast.

mariobehling commented:
@arp95 Please provide a open event web app sample page to make it easier for reviewers to go into detals.

arp95 commented:
@mariobehling is this link what you are expecting?
http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Spark%20Summit%202017/

if not do tell me. this i got through web app generator

mariobehling commented:
@arp95 Thanks. Usually this link will be cleaned after some time. So, best is to deploy this to a github test page. Generating the event as a web app shows that something is not yet working as expected: Tracks, Schedule, Rooms and Speakers are missing in this event. Please check what is missing and ensure the samples are working to enable automatic builds of web app and Android app.

mariobehling commented:
Also, I see that you are generating an event for the Spark conferences, while this issue is about Mozilla.

arp95 commented:
@mariobehling this was just for testing. thanks for clearing out. just doing that now as its clear.

arp95 commented:
@Princu7 please review.

Princu7 commented:
@arp95 Thanks. I tested the samples. Some points:

* In the sessions JSON, the status of the sessions is written as 'draft' instead of 'accepted' or 'completed'. Since the 'draft' means the sessions is not yet confirmed, we don't show them on the generated event site. If you replace the status 'draft' with 'completed', all the pages would appear. I am not entirely sure where the change should be made. On the web app or the event file itself. @mariobehling Please guide. Thanks!!

* Please include the font_color in the track field along with the color. Because the default black text is not readable on the darker backgrounds.

*  On checking, the sponsors JSON file was empty. Please see if any sponsors can be added to the event for more completeness.

* Also, the photo field of the speaker is not defined. Is this intentional? Sorry for not being on the same page.

Please see the above points. Thanks!!

mahikaw commented on sample/MozillaAllHands_17/microlocations:
```diff
@@ -0,0 +1,138 @@
+        "latitude": 37.784878, 
+        "longitude": -122.408965, 
+        "room": ""
+    },
```
Please add location specific coordinates if possible.

mariobehling commented:
>  Since the 'draft' means the sessions is not yet confirmed, we don't show them on the generated event site. If you replace the status 'draft' with 'completed',

@Princu7 @arp95 This is an error in the JSON set. Actually that these sessions show up in the Android app is a bug. Draft sessions should never show up.

mariobehling commented:
@arp95 Please follow the comments of @Princu7 
* Add colors as on Mozilla website
* Add sponsors, e.g. Mozilla
* Add speaker images

mahikaw commented on sample/MozillaAllHands_17/sessions:
```diff
@@ -0,0 +1,2170 @@
+		}, 
+        "signup_url": "", 
+        "speakers": [], 
+        "state": "draft"
```
You could mark them as "Accepted" I believe?

arp95 commented:
@mariobehling making a change in adnroid app too. showing sessions with "draft" doesnt make sense.
yes making the changes he suggested.

mahikaw commented on sample/MozillaAllHands_17/speakers:
```diff
@@ -0,0 +1,764 @@
+        "thumbnail": ""
+    },
+	{
+        "id": 19, 
```
This speaker is repeated 2 times. Please delete one.

mahikaw commented on sample/MozillaAllHands_17/tracks:
```diff
@@ -0,0 +1,230 @@
+    {
+        "id": 455, 
+        "name": "All Hands", 
+        "color": "#2E7D32",  
```
Please add font_color too as suggested by @Princu7 It will make for a better web app.

mahikaw commented on sample/MozillaAllHands_17/tracks:
```diff
@@ -0,0 +1,230 @@
+                "id": 1017, 
+                "title": "Firefox Product & Program/Project Management Homeroom"
+            },
+			{
```
formatting the JSON is a good practice and it won't take much time. Just run it through one of the online formatters once.

mahikaw commented on sample/MozillaAllHands_17/microlocations:
```diff
@@ -0,0 +1,138 @@
+        "id": 15, 
+        "name": "Franciscan C", 
+        "floor": 0, 
+        "latitude": 37.784878, 
```
Please check if they are all at ground level once.

mahikaw commented:
@arp95 while the speaker images are not available on the 2017 site, same speakers were there in 2016 event. You could get their images from the 2016 site.

arp95 commented:
@Princu7 as you can see i have added all the info you said.
only thing bothering me right now is i added the speaker images through google drive links, so they show in the android app but not in the web app link. 
As this is a sample app what should i do?

Link: http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/index.html

Princu7 commented:
@arp95 Okay. Thank you so much. The speaker's page is still not showing up I guess. Also, no images are showing. This might be a problem on the web app. I will look at it as soon as I get some time. Thank you!!

arp95 commented:
@Princu7 oh its a problem in the web app. fine then.
can you please review and approve then.

Princu7 commented:
@arp95 Yeah, the images are having some problems in the web app. The reasons why the sessions are not showing up is that the sessions are marked as 'completed' while the currently accepted keywords are 'accepted' and 'confirmed'. Please change it to one of the above keywords and the speaker image would appear. Thanks.

arp95 commented:
@Princu7 doing it.

arp95 commented:
doing it.

arp95 commented:
@Princu7 is it fine now?

Link: http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/index.html

Princu7 commented:
@arp95 Yes. Works fine. Thanks!! :grinning:

arp95 commented:
@mariobehling please review and merge.

arp95 commented:
@mariobehling just to confirm this. are you saying i have to take all the info from all hands?
because i am taking around 30-40 sessions so that they are enough for a sample.

arp95 commented:
@mariobehling for the images part i am sorry. i added the google drive link of the images but as @Princu7 told there is an error in the web app.

mariobehling commented:
@arp95 Please check how images are handled. Dependency on Google drive is bad! Please compare to the existing FOSSASIA sample.

We want to showcase our app as an alternative to the proprietary sched.org. Therefore it needs to be complete. If you want the CEO of Mozilla to see our work, then we need something that is awesome. Right now we have a below than average kind of thing. As you are member in Google Summer of Code (one of the top tech companies in the world!) we are aiming for an awesome outcome. Why should we work on this project, if we don't want to do something that changes the world? Would be great, if you help to raise the level of this project now.

arp95 commented:
@iamareebjamal 
have a loot over here.
Link: https://github.com/arp95/open-event/tree/mozilla/sample/MozillaAllHands_17

What causing the dates to be rendered in utc format?

iamareebjamal commented:
@arp95 Looks like the new response created some bugs in date parsing. This is related to explicitly adding the timezone + the timezone present in the date itself. It has created an intermediate kind of date.

I think rewriting the date class is the best solution as it is already really complex. Your samples are correct. If you can create a temporary fix in Android app, please do

arp95 commented:
@iamareebjamal i thought checking the web app too but that too is giving the same error.
btw was there any need of using this date format? because adding timezone makes no sense everytime.

iamareebjamal commented:
This is the actual complete ISO8601 spec format. This removes the dependency on external timezone requirements. The new DateUtil class will be a lot simpler, as you'll see

arp95 commented:
you are working on it? or should i take it up?

iamareebjamal commented:
I'm working on it

arp95 commented:
@iamareebjamal so the only advantage is we can get the timezone fro  the date string itself right?
instead of keeping the timezone variable. right?

iamareebjamal commented:
Also, it is ISO standard format, read this -> https://en.wikipedia.org/wiki/ISO_8601

arp95 commented:
@iamareebjamal @Shailesh351 @shreyasnbhat @Princu7 @mahikaw 
finally done. please review.

Link: http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/index.html

Princu7 commented:
@arp95 Solid work. Would it be possible to also add some featured speakers to it? It would look a little better I guess. Thanks!!

arp95 commented:
@Princu7 i have added 40+ speakers and the event has around these many speakers.
what are featured ones btw?

iamareebjamal commented:
Some prominent speakers from the event @arp95

arp95 commented:
which json file should i use for that?

Princu7 commented:
@arp95 Speakers have a boolean field named `featured` I remember. Please cross-check the API though. Thanks!!

arp95 commented:
is that field boolean then? i havent used that field btw.

Princu7 commented:
@arp95 Yes. True or False.

arp95 commented:
oh ok. thanks. looking into it.

Princu7 commented:
https://eventyay.com/api/v1/events/6/speakers
Here is a reference.

arp95 commented:
@Princu7 10 are fine?
how many should we keep?

Princu7 commented:
@arp95 Seems to be a good number :+1:

arp95 commented:
sure sending it right away.

arp95 commented:
Updated link: http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/

iamareebjamal commented:
@arp95 Not sure what is causing it, but only your zip is making the generator throw error which resizing logo, and there are some extra images in the archive as well. Please take a look

Princu7 commented:
@arp95 Thanks. Web app looks better now :+1:  . There are still some points:-

* Please add sponsors images locally too.
* Would it be possible to add speaker descriptions too? Especially the featured ones.
* The font_color on all the tracks is white. It would be better if we could have black font_color for the lighter background colors and retain white for the darker background colors. It would provide a really good contrast and make the app look much better

There are still more issues which are on the web app side. 
* In spite of including the Mozilla banner logo, it doesn't seem to be loading on the event site.
* No placeholder image present on the sessions present on the track and rooms pages when there is no speakers associated with that session.

I will be trying to fix these issues. Please look at the above-mentioned points. Thanks!!

arp95 commented:
@Princu7 thanks will look into them.
@iamareebjamal are you generating an app out of it? because i think images wont be shown as they are added locally.

iamareebjamal commented:
The thing is not about images not being shown, the whole generator is crashing

arp95 commented:
wait let me have a look.
if i am able to get a web app, dont know why not getting an android app.

iamareebjamal commented:
And as of images not displaying, it was fixed in https://github.com/fossasia/open-event-android/pull/1666 and works absolutely fine for other samples like Fossasia 16

![photo6194942385893517289](https://user-images.githubusercontent.com/3874064/27043626-61734072-4fb8-11e7-81f8-67ae1823c110.jpg)
![Uploading photo6194942385893517290.jpg…]()

arp95 commented:
@iamareebjamal this process is getting messier.
there are some things i do for web app, it makes the android app problematic and vice versa
why arent common procedures being followed for both versions. dont get me wrong here but we need to sort out this.

iamareebjamal commented:
The thing is we can't. The build process of both apps and the technology stacks are vastly different. If other samples are working, this should too. And if the problem is actually in the generator, we all are in the same project, we'll fix it. Please don't see the process as a consumer, but as a developer

arp95 commented:
you got me wrong.
i know that we have complete vast different stacks being used. i am talking about the utilization of the api. one uses this field for only online purposes, one other. 
what i mean is the android team is not at all knowing the api properly which i saw by building json files of my own. no one knew draft sessions shouldnt be put. anyways instead of pointing out its better we start working better.
seeing the error. will let you know.

mahikaw commented:
@arp95 I was looking into your sample and the link that you have provided isn't working. Could you please check it once?

arp95 commented:
@iamareebjamal 
i tried three zips, "MozillaAllHands_17", "GoogleIO_17", and "FOSSASIA16" on the app generator. It is giving an error, maybe app generator is not working right now.
Meanwhile you can replace the api link in the config.json with this: 
https://raw.githubusercontent.com/arp95/open-event/allhands/sample/MozillaAllHands_17

Its working fine with this.

arp95 commented:
Also i tried using this api link: "https://raw.githubusercontent.com/arp95/open-event/allhands/sample/FOSSASIA16" in the config.json.

There is no rnedering of images as i said earlier. local images are not being rendered by our app.

![19125459_10211549206880258_1600259772_o](https://user-images.githubusercontent.com/8268392/27045973-6ab36d26-4fc0-11e7-839c-d85ba3f7b4ad.png)

iamareebjamal commented:
@arp95 That's because you did not use the FOSSASIA zip. Please use the zip to include the images.

arp95 commented:
I used the zip actually.
Also the app is working fine if you replace the api link. The generator seems to be giving different results.

iamareebjamal commented:
@arp95 So you got the local images working or not?

arp95 commented:
nope.

arp95 commented:
Also i cant find any error in my json otherwise on repalcing the api link with mine one i would have got an error. In that case the app works fine except the local images problem.

iamareebjamal commented:
Can you please test the generator on machine? Furthermore, I linked my generated apk to show that local images are working. Secondly, if you click yes, on `Download latest schedule`, then it will obviously load the server response and overwrite the local image link

arp95 commented:
ok i will look into that.
but is my apk giving an error and all the other are working? dont seem to find an error

iamareebjamal commented:
Your apk is not giving any error, never said that, I said that your archive(zip) makes the generator crash

arp95 commented:
sorry i meant the zip. strange. looking into it.

Shailesh351 commented:
@arp95 Images are being rendered from local images for FOSSASIA16.zip.
Facing error during generating app from this sample zip while resizing launcher icon as @iamareebjamal  pointed out.

iamareebjamal commented:
@arp95 Got the error. You are supplying JPEG logo, whereas it should be PNG

arp95 commented:
Thanks for looking into.
But i used .jpeg in two files:

1. /images/speakers/mitchell_baker.jpeg
2. /images/speakers/susan_chen.jpeg

and on seeing the files they seem to jpeg.

arp95 commented:
@iamareebjamal 
even the speaker images?

iamareebjamal commented:
No, just the logo. Though Android icons should be png, the error was due to JPEG decoding in PIL because we didn't supply a default quality. We could support JPEG as icon but it is always recommended to use PNG for launcher icons

arp95 commented:
ok sure.

iamareebjamal commented on sample/MozillaAllHands_17/sponsors:
```diff
@@ -0,0 +1,12 @@
+        "level": "1",
+        "url": "https://www.mozilla.org/en-US/firefox/new/",
+        "sponsor_type": "Gold Sponser",
+        "logo": "https://drive.google.com/uc?export=download&id=0B6k_UTO5VOZKQXRDYTQ1aGl2d2c",
```
Please change this to local image too. The app is working fine.

Nice job! Just remove the redundant images from the sample, update the logo and change to local image and you are done

iamareebjamal replied:
Update the zip along this too

arp95 commented:
@iamareebjamal @mahikaw @shreyasnbhat @Shailesh351 
please review now.

arp95 commented:
@Princu7  please review this.

Link: http://opev-webgen-dev.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/index.html

Princu7 commented:
@arp95 Thanks. Web app looking good. I worked on some issues like Mozilla banner not showing on the front page and the UI of the sessions which don't have any speakers on the tracks and rooms pages. The logo still appears too big and we need to make it a little smaller. I am working on it.The changes aren't merged in the official repo. But you can try out the changes I made here:-
http://secure-meadow-20680.herokuapp.com

Also, please change the font_color of the tracks `FP&A` ,  `Strategic Ops`,  `FireFox`,  `Meals with Registered Guests` and  `Meals` to black. It will look a bit more good I guess. Thanks.

arp95 commented:
@Princu7 apart from the font color change is there any other change needed from my end?

mahikaw commented:
@arp95 the android app still isn't working for me. Neither through generator nor if I replace just the API link on android studio.
The web app looks good enough. Just that you have added different font colors for same track colors as well at places. 👍 
@Princu7 I think that different font colors don't look nice. We should stick to one font color only to maintain consistency.

arp95 commented:
app is not working?
@iamareebjamal is it showing some error again?

mahikaw commented:
yes, this is the error on generator that I see.

[2017-06-13 11:23:32,641: ERROR/MainProcess] Task generate.app[f630e99b-d65e-4fc5-8a2f-f9d790889152] raised unexpected: ValueError('Invalid quality setting',)
Traceback (most recent call last):
  File "/Users/tgit/open-event-android/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/utils/libs/request_context_task.py", line 46, in __call__
    result = call()
  File "/Users/tgit/open-event-android/apk-generator/v2/app/utils/libs/request_context_task.py", line 32, in <lambda>
    call = lambda: super(RequestContextTask, self).__call__(*args, **kwargs)
  File "/Users/tgit/open-event-android/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/tasks/__init__.py", line 21, in generate_app_task
    task_handle=self)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/tasks/__init__.py", line 40, in generate_app_task_base
    return generator.generate(should_notify=True)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/generator/generator.py", line 148, in generate
    resize_launcher_icon(self.app_launcher_icon, self.app_working_dir)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/utils/assets.py", line 33, in resize_launcher_icon
    resizer.resize(icon_path)
  File "/Users/tgit/open-event-android/apk-generator/v2/app/utils/libs/asset_resizer.py", line 90, in resize
    return self.resize_image(path, Image.open(path))
  File "/Users/tgit/open-event-android/apk-generator/v2/app/utils/libs/asset_resizer.py", line 122, in resize_image
    im.resize(size, self.image_filter).save(out_file, quality=self.image_quality)
  File "/Users/tgit/open-event-android/lib/python2.7/site-packages/PIL/Image.py", line 1698, in save
    save_handler(self, fp, filename)
  File "/Users/tgit/open-event-android/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 614, in _save
    raise ValueError("Invalid quality setting")
ValueError: Invalid quality setting

iamareebjamal commented:
Let me see

arp95 commented:
@mahikaw the api link works well for me. i think you have to update the app to the latest commit.
as far as generator is concerned, i dont want to go into it again, @iamareebjamal would be better clarifying on this.

mahikaw commented:
The API links don't work in the generator either @arp95 The error is with the sample I think, as per the error log that I sent. Please try and look into it as it will be difficult for @iamareebjamal to correct the samples.

iamareebjamal commented:
@arp95 @mahikaw The generator is working absolutely fine. The app is being generated with all local images included in it

I think you are using old archive @mahikaw Please align to the latest development of the sample

iamareebjamal commented on sample/MozillaAllHands_17/sponsors:
```diff
@@ -0,0 +1,12 @@
+        "level": "1",
+        "url": "https://www.mozilla.org/en-US/firefox/new/",
+        "sponsor_type": "Gold Sponser",
+        "logo": "/images/sponsers/mozilla_sponser.png",
```
Please correct the spelling of `sponsors` or generator won't pick it up

iamareebjamal commented:
@arp95 The api links are working as well

mahikaw commented:
I am not sure how they are working for both of you. I am still unable to run them, specially on the generator. Neither through zip nor API endpoints. On android studio, the images aren't loading.

App on android studio looks as follows:-
![Uploading device-2017-06-13-120054.png…]()
![Uploading device-2017-06-13-120038.png…]()

Generator gives the following error from both API endpoint and zip file.
http://codepad.org/t8zTDumy

Princu7 commented:
@arp95 Nothing more for now. I will check and see if there are some more issue. I know this process of continuous reiteration has been boring and frustrating and I am sorry for that. I hope you don't mind.

@mahikaw Yes. You are right. Actually, earlier we used only the black color for the fonts. But it made the text almost unreadable on some dark backgrounds like dark red and green. So, we switched to binary colors: black and white. That's what we aim for. That the default font_color should be black and we should only switch to white if the font is unreadable on the given track color.

arp95 commented:
@Princu7 no problem. it does seem frustrating at times.

updating to the latest version, please use that if you want any improvments.

iamareebjamal commented:
@mahikaw Here is the app. You can try https://www.dropbox.com/s/8eo5xyfk626kp1s/22841461-bc74-42e8-b30b-cb622e4abfb8.apk?dl=1

And to be clear. Images that are locally added can never load in Android Studio/API link generated apk because they are loaded from zip

mahikaw commented:
@iamareebjamal there is no misunderstanding. My point is that the images should load regardless of how the app is generated. However, can you tell me something about the errors that are coming up while running the generator?

iamareebjamal commented:
There is no error while I generate, for a proof, I have attached the generated APK

iamareebjamal commented:
@mahikaw I am not sure why anyone would use generate via `API endpoint` for a JSON payload ZIP. Can you tell a use case where this will be a valid use case? Either you have a JSON payload zip or as the generator clearly says `API endpoint of event on OpenEvent` and since you don't have the latter, I am really confused how can the generator generate a valid apk for you?

mahikaw commented:
@iamareebjamal of course there is no error on your side because you are able to generate the apk. I am asking if you can tell me about the error i posted so that I am able to look into why I ma unable to generate the app anywhere. Can you do that?

iamareebjamal commented:
APK generated using online generator - https://www.dropbox.com/s/eofhurrq8ynrtgb/641c9ff4-ebda-4a76-ad9c-73be79ad10fe.apk?dl=1

APK generated using local generator - https://www.dropbox.com/s/hgx9yl20o1a5clx/b407e805-4723-44c3-aa2a-5d205990c342.apk?dl=1

iamareebjamal commented:
Can you please tell which sample you are using because I am talking about Mozilla All Hands 2017 latest zip archive. Maybe you could upload the sample zip you are using

mahikaw commented:
Sure, this is the zip I'm using..
[MozillaAllHands_17.zip](https://github.com/fossasia/open-event/files/1070572/MozillaAllHands_17.zip)

iamareebjamal commented:
Again, got no error while generating on both local and online https://droidgen.eventyay.com/ generator

@Shailesh351 can you confirm this too please so that we know that this is not something I can do

mahikaw commented:
@iamareebjamal please understand that no one is blaming you for anything. I am simply seeking your help as a co-developer as I am not able to generate the apk. However, if you don't really know the problem, you can say so. I will try and figure out whatever is wrong with my process of creating the apk. Thanks for your help. 😄

niranjan94 commented:
@iamareebjamal i can confirm the same.
Generated on droidgen: https://droidgen.eventyay.com/static/releases/63adf985-3fed-4161-8477-76ad9f9fb741.apk 
No error on running either.

iamareebjamal commented:
@mahikaw I'm sorry. I am seemingly harsh but as you and @arp95 get frustrated by constant change requirements, I get too because I well invested my time yesterday in these samples and checking every sample of Open Event to ensure there is nothing wrong with the generator. I'm really sorry for any inconvenience caused but it really confused me because the same zip can't cause errors for different people on the online generator. Again sorry for any inconvenience caused. I too, just want the project to be moving forward in a way that everyone is on the same page and focused on issues which are common

mahikaw commented:
Yes, that is my point of confusion too, that's what I was trying to figure out because all the other samples work, inlcuding my samples of OSCON and others. However, it's okay if it's working for everyone else 👍

Shailesh351 commented:
@mahikaw @arp95 @iamareebjamal Faced no error while generating app and all the images(speakers/sponsors/logo)  are rendering offline.

Still there is little bug i found. If there is already folder in assets like `images/speakers` or `images/sponsors`.
Then it will give error : File already exists.

At following line
`shutil.copytree(speakers_path, self.get_path("app/src/main/assets/images/speakers/"))`
and
`shutil.copytree(sponsors_path, self.get_path("app/src/main/assets/images/sponsors/"))`

As per https://docs.python.org/2/library/shutil.html  destination in copyTree must not already exist.

@mahikaw  check if you have this folders in assets or not. If there, remove them for now for testing.

arp95 commented:
@iamareebjamal @Shailesh351 
is there any changes i have to make for the android app then? If not please approve.

arp95 commented:
@Princu7 is there anything needed now for the web app? If not please approve.

Shailesh351 commented on sample/MozillaAllHands_17/tracks:
```diff
@@ -0,0 +1,971 @@
+        "id": 455, 
+        "name": "All Hands", 
+        "color": "#E74C3C",
+		"font_color": "white",
```
Shouldn't it be in hex ?

arp95 replied:
thanks for pointing out. changing it.

Shailesh351 commented:
@arp95 Just for suggestion try to format all the files to `4 space tab` format so that it can be more readable. 
https://jsonformatter.org/

arp95 commented:
@Shailesh351
actually it is formatted on sublime in which i work. will see why its not formatted when i load files on github.

arp95 commented:
@Shailesh351 @shreyasnbhat @mahikaw 
please review the app.

@Princu7 
please review the web app so that it can be taken forward.

Princu7 commented:
@arp95 Looks good. Great work. Just one simple change:
* There is a picture of a speaker named `madhava_enros.jpg` in the images folder but it is specified in Upper case in the speakers JSON file and it is not rendering properly because of it. Please check that. Thanks!!

arp95 commented:
done @Princu7

arp95 commented:
@mariobehling you can have a look now.

Link: https://arp95.github.io/MozillaAllHands_17/

shreyasnbhat commented:
@arp95 Are the speaker images like for Travis Blow,Mozilla intended or wouldn't it be better to have a placeholder instead?

arp95 commented:
@shreyasnbhat
i was asked to collect images from sites like linkedin nd all. Travis Blow for example has a LEGO toy photo for his linked profile so i kept it there as the white background doesnt make the app look good.
Also placeholders are encouraged and i have written on Gitter on why they dont have it by default.

arp95 commented:
@shreyasnbhat 
its better to remove it right? yeah it looks odd.

arp95 commented:
@mahikaw @shreyasnbhat 
made the changes.

arp95 commented:
@mariobehling @niranjan94 
please review.

Link: https://arp95.github.io/MozillaAllHands_17/index.html

Princu7 commented:
@arp95 Looks Good. The background image doesn't appear but that is due to an issue in the web app.
You can try generating through this test-server. I have made many fixes in it. Thanks!!
http://secure-meadow-20680.herokuapp.com.

arp95 commented:
sure. providing the link using that. @Princu7

arp95 commented:
@Princu7 is this fine?

Link: http://secure-meadow-20680.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/

Also on downloading i am not getting a zip so dont know how to deploy it on a github test page.

arp95 commented:
@mariobehling @niranjan94 
please review.

Link:  https://arp95.github.io/MozillaAllHands_17/

mariobehling commented:
@elioqoshi Is this how the logo of Mozilla should be displayed? https://arp95.github.io/MozillaAllHands_17/

elioqoshi commented:
Taking a look at this now @mariobehling and will give feedback in a bit.

Regarding the logo: Yes it's fine, but it has some white border, please remove that. You might use this:
![mozilla eventyay](https://user-images.githubusercontent.com/5436686/27179526-4a117fde-51cf-11e7-8c97-42fc3ac9dc25.png)

mariobehling commented:
Some images are of too low quality. Please go through the images and find better ones where necessary. E.g. see below:
![screenshot from 2017-06-15 13-35-30](https://user-images.githubusercontent.com/1583873/27179601-9cc2902e-51cf-11e7-995e-f30f77710770.png)

elioqoshi commented:
Furthermore: Fix the Sponsor section or remove it altogether.

In the schedule, please check colors for selected and inactive buttons. When selected it lacks contrast 
![eventyay selectors](https://user-images.githubusercontent.com/5436686/27179648-c7778e14-51cf-11e7-96e8-6fd971946231.jpg)

elioqoshi commented:
![active](https://user-images.githubusercontent.com/5436686/27179668-e5aea250-51cf-11e7-953a-39de6da668b6.jpg)

The List and Calendar View are quite confusing to see which one is active. Adjust the colors and contrast of the buttons to make it more clear which one is active and which not

Princu7 commented:
@eloqoshi Thanks for the review. The pull requests for those issues have already been sent. They will get merged soon :)

arp95 commented:
@Princu7 
arent these issues from web app side?
regarding the images replacing them with better ones.

elioqoshi commented:
Also, when expanding a session/room the animation flickers at the very end of the expansion. Can you fix that? (I hope you get what I mean)

Princu7 commented:
@arp95 Hey. I have deployed a test server and enabled zip download:
http://secure-meadow-20680.herokuapp.com
It fixes many of the current issues. Please deploy the web app using it. Thanks :)

mariobehling commented:
@arp95 I am also unclear why you are not following best practices in this PR. Please add the issue that is getting solved into the description.

arp95 commented:
@mariobehling 
doing the requested changes.

elioqoshi commented:
![1](https://user-images.githubusercontent.com/5436686/27180177-6444b774-51d2-11e7-90cf-f37b872aef5a.jpg)

In the speakers section it is quite not user friendly to list sessions a speaker has underneath their photo like here:

![1](https://user-images.githubusercontent.com/5436686/27180195-7a4e11b4-51d2-11e7-8f1d-31131fd95792.jpg)

I'd propose a change to use the width of the interface and darken out all the other speakers like this (very rough sketch)

![2](https://user-images.githubusercontent.com/5436686/27180210-94a33f58-51d2-11e7-89d9-67cdc962abdb.jpg)

arp95 commented:
@iamareebjamal @Shailesh351 @mahikaw @shreyasnbhat 
have added the entire info of the sample now. can you guys check the app again in case of any errors on my side.

mariobehling commented:
@arp95 Please make it easier for reviewers and also always share a sample web app in the description. This one https://arp95.github.io/MozillaAllHands_17/ does not show a newer version.

arp95 commented:
@Princu7 
have a look at it now.
Link: http://secure-meadow-20680.herokuapp.com/live/preview/arpitdec5@gmail.com/Mozilla%20All%20Hands%202017/index.html

arp95 commented:
@mariobehling 
added the link above.

mariobehling commented:
As Herokuapps get purged after a short while, they are not suitable for showcases. Please use the same github pages URL until this issue is solved.

It seems like you are not reading comments in detail. Still a lot of the same issues as before. The long comment queue on this issue is honestly unacceptable. Please provide more quality work:

* Speaker and sessions are still missing.
* Correct titles and short bio still missing.
* Background image not in suitable size.
* Many speaker images still of too low quality.

arp95 commented:
@mariobehling 
i have added the entire information. the speaker missing are ones which have no sessions. the web doesnt show that.

i did my best finding the speaker iamges from the net but will try more if you say.

arp95 commented:
Test Page Link: https://arp95.github.io/MozillaAllHands_17/ 
Original Event Link: https://sanfranciscoallhandsjune2017.sched.com/

1. Have added all the speaker images and their short bio ( 4 are of low quality, i have opened a separate issue for them #164 )

2. All sessions and speakers are added:
Number of tracks: 15
Number of sessions: 372
Number of speakers: 58
( original event has 59 speakers but that extra speaker has name "Chris Beard/Sean White" whose individual names i have already included, so their sessions are added in each of them)

3. Background image quality looks better now.

@mariobehling please review now.

elioqoshi commented:
It's kinda pointless to display "Mozilla" as the organisation for all speakers as that is self understood. I'd suggest to list the Team everyone represents instead.

arp95 commented:
@elioqoshi 
sorry for replying late. yeah it becomes quite pointless. conveying this to the web app team and making the required changes. anything else?

arp95 commented:
@mariobehling 
have updated the link.

Link: https://arp95.github.io/MozillaAllHands_17/

mariobehling commented:
Ok, working now. Hope we can get other events faster. Please keep checking if we have everything. Thanks!

elioqoshi commented:
Can we remove the sponsors please?

elioqoshi commented:
And Brianna should be the contact here not support@mozilla.com

mariobehling commented:
@elioqoshi These are placeholders. Right now this data will be public. So, we should not put Brianna's email public I guess.

The sponsors data is a placeholder. This section is rendered automatically. I opened an issue to solve this: https://github.com/fossasia/open-event-webapp/issues/1366
